### PR TITLE
Add `remark-hexo` to list of plugins

### PR DESCRIPTION
**Pull request by bennycode**
### Initial checklist

*   [x] I read the support docs 
*   [x] I read the contributing guide 
*   [x] I agree to follow the code of conduct 
*   [x] I searched issues and couldn’t find anything (or linked relevant results below) 
*   [x] If applicable, I’ve added docs and tests

### Description of changes

I added a link to the [remark-hexo](https://github.com/bennycode/remark-hexo) plugin which I built.

### CODE REVIEW

File: doc/plugins.md [hunk 1]
```diff
@@ -316,6 +316,8 @@ The list of plugins:
     — configure remark w/ YAML
 *   🟢 [`remark-code-title`](https://github.com/kevinzunigacuellar/remark-code-title)
     — add titles to code blocks
+*   🟢 [`remark-hexo`](https://github.com/bennycode/remark-hexo)
+    — renders [Hexo tags](https://hexo.io/docs/tag-plugins)
```
Comment (wooorm) [inline doc/plugins.md:320]: Can you sort it in the list? Thanks

(I see the one above is also wrongly places, meh, anyway, feel free to ignore that!)

Comment (bennycode) [reply doc/plugins.md]: No problem! I sorted mine as well as the one from above (remark-code-title) :)

**Comment by ChristianMurphy** [comment]
Thanks @bennycode!
A few thoughts:
1. running this as a transformer, creates a caveat that it may not work with other plugins which look for the `{` symbol (like MDX). If that is needed, you may need to parse at the micromark level: see https://github.com/micromark/micromark#extending-markdown and https://github.com/micromark/micromark#creating-a-micromark-extension
2. If it continues as a transformer, consider leveraging https://github.com/syntax-tree/mdast-util-find-and-replace
3. Is it intentional that the code does not have a license? Without a license it is not open source and others cannot use it. Consider a permissive license like MIT or Apache 2 for more adoption, or GPL or SSPL for stronger copyleft.
4. In your tsconfig, consider setting module resolution to `node16` to ensure compatibility with TypeScript with ESM

**Comment by ChristianMurphy** [comment]
> What is so wrong about my technique? 

It's not necessarily "wrong".
I intuit that there may be some edge cases you are not considering:

1. what happens when `remark-hexo` is used with other parser plugins like `remark-mdx`? `remark-mdx` is a parser plugin, it runs at the very beginning, when it sees `{` it inteprets it as a JSX block, it does not know about `remark-hexo` and will fail the document because `{%` is invalid JSX. Alternatively, you can document "this may not work with `remark-mdx` or other plugins that use `{}`".
2. There could be text before or after the Hexo template, currently that text will be erased/replaced. `mdast-util-find-and-replace` will keep the other surrounding text. (you could also do this with `unist-util-visit` but it is more involved)

**Comment by bennycode** [comment]
Ah, thanks for the detailed info. I am using the `remark-hexo` plugin myself in an Astro blog (which uses Remark & Rehype) and is it was working fine I was assuming there is nothing wrong with it. Need to double check what happens with MDX files as I was only using the `{` syntax in MD files.

**Comment by ChristianMurphy** [comment]
Again, you don't _have_ to make them work together.
MDX and Hexo feel more like alternatives than something to use together.
But it could be something to document as an intended incompatibility to avoid people logging issues, since MDX is something commonly used with Astro.

**Review by remcohaszing — APPROVED** [review]

**Review by wooorm — APPROVED** [review]

**Comment by wooorm** [comment]
Thanks!

**Pull request merged**